apps/jubmoji-quest/src/pages/api/quests.ts

```diff
@@ -0,0 +1,117 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import prisma from "@/lib/prisma";
+import { getRandomNullifierRandomness } from "jubmoji-api";
+import { bigIntToHex } from "babyjubjub-ecdsa";
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  if (req.method === "GET") {
+    try {
+      const quests = await prisma.quest.findMany({
+        select: {
+          id: true,
+          name: true,
+          description: true,
+          tags: true,
+          startTime: true,
+          endTime: true,
+          proofType: true,
+          proofParams: true,
+          // Selecting specific fields from related models
+          prerequisiteCards: {
+            select: {
+              index: true,
+            },
+          },
+          collectionCards: {
+            select: {
+              index: true,
+            },
+          },
+        },
+      });
+
+      const transformedQuests = quests.map((quest) => {
+        const prerequisiteIndices = quest.prerequisiteCards.map(
+          (card) => card.index
+        );
+        const collectionIndices = quest.collectionCards.map(
+          (card) => card.index
+        );
+        return {
+          ...quest,
+          prerequisiteCardIndices: prerequisiteIndices,
+          collectionCardIndices: collectionIndices,
+          // Remove the original fields from the response object
+          prerequisiteCards: undefined,
+          collectionCards: undefined,
+        };
+      });
+      res.status(200).json(transformedQuests);
+    } catch (error) {
+      res.status(500).json({ error: "Unable to fetch quests" });
+    }
+  } else if (req.method === "POST") {
+    try {
+      // Here, we're assuming the body contains the quest details.
+      // You should validate these details before attempting to create a quest in your database.
```

Comment:
So this would be plugged into some airtable type setup?

Reply:
We also need some spec here for "data validation" to avoid this endpoint getting abused

Reply:
Yeah, this could be airtable with manual input/some hook. Or we could expose a frontend that creates pending quests which need our approval. If we want creating quests to be fully self serve, we have to simplify the language/process around creating a quest.